### PR TITLE
Pv2 838 dlock 03 validator

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

File: src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs [hunk 1]
```diff
@@ -135,41 +136,61 @@ public async Task GivenCourseValidationDataLockIsReturnedMapOnlyValidEarningPeri
...
+                }
+            };
+
             onProgValidationMock
                  .Setup(x => x.ValidatePeriods(apprenticeships[0].Uln, 
                     It.IsAny<List<PriceEpisode>>(),
                     It.IsAny<OnProgrammeEarning>(),
-                    It.IsAny<List<ApprenticeshipModel>>()))
-                .Returns(() => (new List<ValidOnProgrammePeriod>
+                    It.IsAny<List<ApprenticeshipModel>>(), TODO))
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs]: Look like a merge Issue

File: src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs [hunk 2]
```diff
@@ -81,21 +81,19 @@ public async Task GivenNoDataLockErrorAllEarningPeriodsShouldBePayableEvent()
...
-                .Returns(() => (new List<ValidOnProgrammePeriod>
-                {
-                    new ValidOnProgrammePeriod
-                    {
-                        Apprenticeship = apprenticeships.FirstOrDefault(),
-                        Period = earningEvent.OnProgrammeEarnings.FirstOrDefault()?.Periods.FirstOrDefault()
-                    }
-                }, new List<InvalidOnProgrammePeriod>()))
-                .Verifiable();
+                                                apprenticeships, TODO))
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs]: Look like a merge Issue remove TODO

File: src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs [hunk 3]
```diff
@@ -135,41 +136,61 @@ public async Task GivenCourseValidationDataLockIsReturnedMapOnlyValidEarningPeri
...
+                }
+            };
+
             onProgValidationMock
                  .Setup(x => x.ValidatePeriods(apprenticeships[0].Uln, 
                     It.IsAny<List<PriceEpisode>>(),
                     It.IsAny<OnProgrammeEarning>(),
-                    It.IsAny<List<ApprenticeshipModel>>()))
-                .Returns(() => (new List<ValidOnProgrammePeriod>
+                    It.IsAny<List<ApprenticeshipModel>>(), null))
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs]: Why are we passing in null can you please use it.Any() makes it more clearer

File: src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs [hunk 4]
```diff
@@ -99,10 +109,10 @@ public void MatchesApprenticeship()
 
 
             var periods = mocker.Create<OnProgrammePeriodsValidationProcessor>().ValidatePeriods(1, priceEpisodes,
-                earning, apprenticeships);
+                earning, apprenticeships, null);
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs]: Please pass in an object instead oof null we will always have a LearningAim

File: src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs [hunk 5]
```diff
@@ -118,12 +128,12 @@ public void DataLocksAreReturnedInInvalidPeriods()
                 .Returns(() => new CourseValidationResult { MatchedPriceEpisode = new ApprenticeshipPriceEpisodeModel { Id = 90 } });
             mocker.Mock<ICourseValidationProcessor>()
                 .Setup(x => x.ValidateCourse(It.Is<DataLockValidationModel>(model => model.Apprenticeship.Id == 2)))
-                .Returns(() => new CourseValidationResult { DataLockErrors = new List<DataLockErrorCode> { DataLockErrorCode.DLOCK_03 } });
+                .Returns(() => new CourseValidationResult { DataLockFailures = dataLockFailures });
 
             var periods = mocker.Create<OnProgrammePeriodsValidationProcessor>().ValidatePeriods(1, priceEpisodes,
-                earning, apprenticeships);
+                earning, apprenticeships, null);
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs]: lease pass in an object instead oof null we will always have a LearningAim

File: src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs (hunk 3 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Services/DataLockProcessorTests.cs]: Changed

File: src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs (hunk 4 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs]: Changed

File: src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs (hunk 5 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.DataLocks.Domain.UnitTests/Services/CourseValidation/OnProgrammePeriodsValidationProcessorTests.cs]: Changed

**Pull request merged**